extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -120,11 +121,10 @@ export class GoogleAuth {
     if (save || !scopes) { // if tokens will be saved (meaning also scopes should be pulled from storage) or if no scopes supplied
       scopes = await GoogleAuth.apiGoogleAuthPopupPrepareAuthReqScopes(acctEmail, scopes || GoogleAuth.defaultScopes());
     }
-    // below codes are all temporary codes and need cleanup if we want to proceed this way
     const authRequest: AuthReq = { acctEmail, scopes, csrfToken: `csrf-${Api.randomFortyHexChars()}` };
     const authUrl = GoogleAuth.apiGoogleAuthCodeUrl(authRequest);
     const authWindowResult = await OAuth2.webAuthFlow(authUrl);
-    const authRes = await GoogleAuth.getAuthRes({ acctEmail, save, authWindowResult });
+    const authRes = await GoogleAuth.getAuthRes({ acctEmail, save, state: GoogleAuth.apiGoogleAuthStatePack(authRequest), authWindowResult });
```

Comment:
It seems you are probably creating `state` twice, once for request and once for verification?

It would be more natural to have a method `GoogleAuth.newAuthRequest(...) and then `state` would be one of the values in `AuthReq`. Then `getAuthRes` would have `expectedState: authRequest.state`.

It would better communicate the intetion for csrf protection, which is "I decide the value in the beginning and it must come back the same". Computing it twice obscures that intention.